bundles/org.openhab.binding.warmup/src/main/java/org/openhab/binding/warmup/internal/discovery/WarmupDiscoveryService.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.warmup.internal.discovery;
+
+import static org.openhab.binding.warmup.internal.WarmupBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.warmup.internal.handler.MyWarmupAccountHandler;
+import org.openhab.binding.warmup.internal.handler.WarmupRefreshListener;
+import org.openhab.binding.warmup.internal.model.query.LocationDTO;
+import org.openhab.binding.warmup.internal.model.query.QueryResponseDTO;
+import org.openhab.binding.warmup.internal.model.query.RoomDTO;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+
+/**
+ * The {@link WarmupDiscoveryService} is used to discover devices that are connected to a My Warmup account.
+ *
+ * @author James Melville - Initial contribution
+ */
+@NonNullByDefault
+public class WarmupDiscoveryService extends AbstractDiscoveryService
+        implements DiscoveryService, ThingHandlerService, WarmupRefreshListener {
+
+    private @Nullable MyWarmupAccountHandler bridgeHandler;
+    private @Nullable ThingUID bridgeUID;
+
+    public WarmupDiscoveryService() {
+        super(DISCOVERABLE_THING_TYPES_UIDS, 5, false);
+    }
+
+    @Override
+    public void activate() {
+        super.activate(null);
+    }
+
+    @Override
+    public void deactivate() {
+        super.deactivate();
+    }
```

Comment:
Can these be removed?

Reply:
I copied this pattern from this binding https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.draytonwiser/src/main/java/org/openhab/binding/draytonwiser/internal/discovery/DraytonWiserDiscoveryService.java, and I think this part of the code was rewritten by @Hilbrand (who also suggested previous changes to how I'd written the discovery service in #7998). I have to admit I'm not 100% up to speed with these services, so a bit of this was trial and error on my part.

It looks like I can remove the `activate()` method and the discovery still works but not the `deactivate()` method, which gives the compilation error `The inherited method AbstractDiscoveryService.deactivate() cannot hide the public abstract method in ThingHandlerService`, so I think I'm making the code compile (`deactivate()` in `AbstractDiscoveryService` is protected, but public in the `ThingHandlerService`), but because background discovery isn't enabled, the method might as well just be empty (and `super.activate(null)` wouldn't be doing anything either). Does this sound right?

Reply:
That's fine.